test/loguru_test.cpp

```diff
@@ -33,34 +37,62 @@ void sleep_ms(int ms)
 	std::this_thread::sleep_for(std::chrono::milliseconds(ms));
 }
 
+void test_thread_names_strong()
+{
+	std::random_device rd;  //Will be used to obtain a seed for the random number engine
+	std::mt19937 gen(rd());
+
+	std::vector<std::thread> threads;
+	for(long long i=1; i<10000; i++) {
+		try {
+			threads.emplace_back(std::thread([i](std::mt19937::result_type sleep_time_in_ms){
+						// name may exceed 16 characters
+						std::string name = std::to_string(std::numeric_limits<long long>::max() / (i*i));
+						loguru::set_thread_name(name.c_str());
+						char thread_name[LOGURU_THREADNAME_WIDTH+1];
+						std::this_thread::sleep_for(std::chrono::milliseconds(sleep_time_in_ms));
+						loguru::get_thread_name(thread_name, sizeof(thread_name), false);
+						CHECK_F(name.compare(0, LOGURU_THREADNAME_WIDTH, thread_name) == 0, "Thread name error!");
+					}, gen()%100));
+		} catch (std::system_error& ) { // if thread creation failed, try again.
+			i--;
+		}
+	}
+	for(auto & t : threads)
+		t.join();
+}
 void test_thread_names()
 {
 	LOG_SCOPE_FUNCTION(INFO);
 
 	{
-		char thread_name[17];
+		char thread_name[LOGURU_THREADNAME_WIDTH+1];
 		loguru::get_thread_name(thread_name, sizeof(thread_name), false);
 		LOG_F(INFO, "Hello from main thread ('%s')", thread_name);
+		CHECK_F(std::string("main thread").compare(0, std::string::npos, thread_name) == 0, "Thread name error!");
```

Comment:
Why not `CHECK_F(std::string("main thread") == thread_name);` or `CHECK_F(strcmp(thread_name, "main thread") == 0);` here and elsewhere?